src/main/python/pybuilder/plugins/python/coverage_plugin.py

```diff
@@ -306,3 +360,32 @@ def _get_system_assets():
                 break
 
     return packages, modules
+
+
+class ModuleCoverageReport(object):
+    def __init__(self, coverage_analysis):
+        self.lines_total = sorted(coverage_analysis.statements)
+        self.lines_excluded = sorted(coverage_analysis.excluded)
+        self.lines_missing = sorted(coverage_analysis.missing)
+        numbers = coverage_analysis.numbers
+        self.n_lines_total = numbers.n_statements
+        self.n_lines_excluded = numbers.n_excluded
+        self.n_lines_missing = numbers.n_missing
+        self.n_lines_covered = self.n_lines_total - self.n_lines_missing
+        self.n_branches = numbers.n_branches
+        self.n_branches_partial = numbers.n_partial_branches
+        self.n_branches_missing = numbers.n_missing_branches
+        self.n_branches_covered = self.n_branches - self.n_branches_missing
+        self.n_branches_partial_covered = self.n_branches - self.n_branches_partial
+
+        if self.n_lines_total == 0:
+            self.code_coverage = 0
```

Comment:
@mriehl same here: no lines whatsoever = no coverage?
1 line is either covered or not. 0 lines, to my mind, are always covered. Old behavior is `0`.